### DISCUSSION TITLE
Snapshot mode

### DISCUSSION THREAD

**Discussion by donagrejja** [tsightler/ring-mqtt#597] [Support Q&A]
Hi.

If I change snapshot mode to disabled and then use camera.snapshot to do snapshots based on automations on the entities I need.

Will that work ? will it actually take a snapshot when called or is the whole function disabled when choosing disabled in snapshot mode ?

Thanks.

**Comment by donagrejja** [comment]
When I've tested it doesn't seem like camera.snapshot service takes a snapshot at all ? Are snapshots only controlled by the timers / events ?

**Comment by donagrejja** [comment]
Snapshots interval also seems to be highly unreliable. At least for the video doorbell. But if I open the Ring app on the phone then it force refreshes all the snapshots it seems.

**Comment by tsightler** [comment]
> If I change snapshot mode to disabled and then use camera.snapshot to do snapshots based on automations on the entities I need.

No, this will not work as the camera.snapshot function does nothing but grab a jpeg of the current images in the MQTT camera entity, it doesn't inform the underlying entity to update the snapshot.  The camera.snapshot feature is designed to take a still image snapshot of a camera stream.

> When I've tested it doesn't seem like camera.snapshot service takes a snapshot at all ? Are snapshots only controlled by the timers / events ?

Snapshot on motion is broken right now due to Ring making changes to the API a few days ago, there is a fix in the dev branch but I haven't had time to push a new version.

> Snapshots interval also seems to be highly unreliable. At least for the video doorbell. But if I open the Ring app on the phone then it force refreshes all the snapshots it seems.

There's no reason it should be highly unreliable.  This feature has been around for years and I use it on 8 cameras with 30 second snapshots with basically 100% reliability and there have been no other reports of issues from other users, and, trust me, this project has well more than 6000 users and people are not hesitant to report issues.  If you have logs showing that snapshots are "unreliable" feel free to share them.

**Reply by donagrejja** [reply]
Thanks for the reply =) What I meant with unreliable was that the video doorbell's snapshot frequency is not according to the timer. The spotlight cameras work as they should though with the timer.

**Reply by donagrejja** [reply]
I can see now that it happened with the doorbell and the cameras aswell. doorbell seems more affected though.
![image](https://user-images.githubusercontent.com/14802028/229107206-6bc88782-58fe-404b-ac58-fe295ac95f74.png)
![image](https://user-images.githubusercontent.com/14802028/229107299-79608116-11ec-49e0-b722-e5ae8a6d6a92.png)

**Reply by donagrejja** [reply]
But those timestamps seems to match the windows where I won't get motion alerts either. So it's probably some kind of connectivity issue with Ring's service.

**Reply by donagrejja** [reply]
I've tried checking the addon logs but I can't see that far back in them.

**Reply by tsightler** [reply]
Are you just using the UI?  It's almost impossible to get enough logs in the HA UI to perform any troubleshooting.  You'll need to install the SSH & Web Terminal addon and use the docker command line if you want to get reasonable logs.

**Reply by donagrejja** [reply]
Tried it but that doesn't give me access to docker. I'm running HassOS. I'll try to get access to the logs =)

**Reply by tsightler** [reply]
Yes, it does, but you have to disable Protection mode and then you can run standard docker commands as much as you'd like.  The process for outputting to a file is documented here:

https://github.com/tsightler/ring-mqtt-ha-addon/blob/main/GET-LOGS.md

**Reply by donagrejja** [reply]
Thanks. I was looking for protection mode but it wasn't there. turned out i was using the wrong addon :D

**Reply by donagrejja** [reply]
2023-03-31T06:16:28.065Z ring-mqtt [32m[Front Door] [39mError: Snapshot for Front Door (doorbell_v4 - Doorbell 2) failed to refresh after 15 seconds.  This is normal behavior since this camera is unable to capture snapshots while streaming
2023-03-31T06:16:28.065Z ring-mqtt [32m[Front Door] [39mFailed to retrieve updated snapshot
2023-03-31T06:16:28.070Z ring-mqtt [32m[Garden] [39mError: Snapshot for Garden (stickup_cam_v4 - Spotlight Cam) failed to refresh after 15 seconds.  This is normal behavior since this camera is unable to capture snapshots while streaming

The log is full of these types of messages during the time where there are no snapshots taken. But there is no motion or streaming logged in the ring app in that timeframe.

**Reply by donagrejja** [reply]
I wonder if it's motion that isn't registered because it's outside of the "motion zone". Then the cameras would be streaming to rings webservice to parse the video based on my settings.

**Reply by donagrejja** [reply]
But I also see these messages during aproximately the same periods:
2023-03-31T05:24:04.300Z ring-mqtt [32m[Front Door] [39mHTTPError: Response code 400 (Bad Request)
2023-03-31T05:24:04.300Z ring-mqtt [32m[Front Door] [39mFailed to retrieve updated snapshot

**Reply by tsightler** [reply]
The error message itself is from getSnapshot() function in underlying ring-client-api, not ring-mqtt itself.  That code makes an assumption that, if it is a camera operating in "low-power" mode (i.e. not line powered) that the reason for the timeout is due to the fact that the camera is streaming.  This could be an incorrect assumption as there could be other reasons for the timeout, network connectivity issues, Ring services issues, etc.

As it seems to be happening for both cameras at the same time, that reinforces the idea that it is likely a network related issues.  I mean, it could possibly be a Ring service issue, but I've found the snapshot API to be pretty reliable (can't same the same about other parts of Ring API, although it seems to be improving as the incident rate this year so far has been low vs previous years).

**Reply by tsightler** [reply]
> I wonder if it's motion that isn't registered because it's outside of the "motion zone". Then the cameras would be streaming to rings webservice to parse the video based on my settings.

But this should still show in the Ring logs, you said there were no events in Ring app event logs.

**Reply by tsightler** [reply]
> But I also see these messages during aproximately the same periods:
> 2023-03-31T05:24:04.300Z ring-mqtt �[32m[Front Door] �[39mHTTPError: Response code 400 (Bad Request)
> 2023-03-31T05:24:04.300Z ring-mqtt �[32m[Front Door] �[39mFailed to retrieve updated snapshot

What are the lines immediately before these lines.  This appears to be a motion based snapshots vs an interval snapshot.  As noted above, those are broken right now due to a change Ring made to the API last weekend.  The fix is in the dev branch for this.

Again, everything is logged, and you must look at the logs all together to put the peices together.  The logs for an interval snapshot are different from motion snapshot and you can easily tell them apart.

**Reply by donagrejja** [reply]
Ok. I will look into it =) Thank you for you replies and information.

**Reply by donagrejja** [reply]
Updated to latest version. Cameras still seem to stop updating snapshots and events at some point or multiple during the day. I have to restart the addon to make all cameras start updating event and snapshots again. I'll post some logs when I've checked them.

**Reply by donagrejja** [reply]
Addon restart didn't help either. Something is fishy.

**Reply by donagrejja** [reply]
After some fiddling with both app and addon it seems to have started working again. Sadly the logs seem to have been cleared upon addon restart so I can't see any errors from when they stopped updating.

**Reply by donagrejja** [reply]
Funny thing though. 2 out of 3 cameras stopped updating events and snapshots around 5-7 this morning. but the 1 camera have been working all day. but when I tested the cameras that wasn't updating they worked and started to work after I manually did a live stream a few times.
